### DISCUSSION TITLE
Receive(ChMux) error when receiving messages over a remote channel

### DISCUSSION THREAD

**Discussion by itamarst** [ENQT-GmbH/remoc#14] [Q&A] [answered]
In my server process, I have some code that reads from a remoc channel. The `unwrap()` is there for debugging purposes, for now, eventually it'll fail more gracefully.

```rust
while let Some(command) = rx.recv().await.unwrap() {
    // ... do something with the command
}
```

I did an end-to-end unit test, client to server—that worked.

However, in my real program, I get an error: `called Result::unwrap() on an Err value: Receive(ChMux)`. What does this mean?

**Comment by surban** [comment, marked as answer]
It means that the underlying connection has been disconnected or a protocol error occurred. Usually this happens when one endpoint closes the connection while there are still channels open. It may be safe to ignore on your server side, since it usually just means that the client has disconnected.
